kani-driver/src/metadata.rs

```diff
@@ -151,13 +151,28 @@ impl KaniSession {
     }
 }
 
+/// Sort harnesses such that for two harnesses in the same file, it is guaranteed that later
+/// appearing harnesses get processed earlier.
+/// This is necessary for the concrete playback feature (with in-place unit test modification)
+/// because it guarantees that injected unit tests will not change the location of to-be-processed harnesses.
+pub fn sort_harnesses(harnesses: &[HarnessMetadata]) -> Vec<HarnessMetadata> {
```

Comment:
Can you use https://doc.rust-lang.org/std/primitive.slice.html#method.sort?

Reply:
Originally, I called sort directly on the slice (I think this is what you are recommending). Then, @adpaco-aws [commented that I should return a new vector](https://github.com/model-checking/kani/pull/1558#discussion_r953961274), so I changed the code. These two recommendation seem to conflict. Which one would you like me to do?

Reply:
Oh, sorry, I didn't see that comment.

You can leave it as is.